src/main/scala/com/sumologic/sumobot/http_frontend/SumoBotHttpServerOptions.scala

```diff
@@ -0,0 +1,84 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.sumologic.sumobot.http_frontend
+
+import com.sumologic.sumobot.http_frontend.authentication.{HttpAuthentication, Link, NoAuthentication}
+import com.typesafe.config.{Config, ConfigFactory}
+
+import scala.collection.JavaConverters._
+import scala.util.{Success, Try}
+
+case class SumoBotHttpServerOptions(httpHost: String, httpPort: Int,
+                                    origin: String, authentication: HttpAuthentication,
+                                    title: String, description: Option[String],
+                                    links: Seq[Link])
+
+object SumoBotHttpServerOptions {
+  val DefaultOrigin = "*"
+  val DefaultAuthentication = new NoAuthentication(ConfigFactory.empty())
+  val DefaultTitle = "Sumobot-over-HTTP"
+
+  def fromConfig(config: Config): SumoBotHttpServerOptions = {
```

Comment:
I bet this could be avoided, if we had a (case) class with just properties called the same as the fields in the config. But it can stay.